llvm/docs/LangRef.rst

```diff
@@ -19209,6 +19209,35 @@ will be on any later loop iteration.
 This intrinsic will only return 0 if the input count is also 0. A non-zero input
 count will produce a non-zero result.
 
+'``llvm.experimental.vector.partial.reduce.add.*``' Intrinsic
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Syntax:
+"""""""
+This is an overloaded intrinsic.
+
+::
+
+      declare <4 x i32> @llvm.experimental.vector.partial.reduce.add.v2i32.v8i32(<8 x i32> %in)
+      declare <4 x i32> @llvm.experimental.vector.partial.reduce.add.v4i32.v16i32(<16 x i32> %in)
```

Comment:
Is it best for these intrinsics to take a single operand?  I kind of see them more like binary operators whose input and output operands are less restrictive.

The intent being we're doing a partial reduction where currently LoopVectorize achieves this via an `add` (or other binop) instruction that is too restrictive (i.e. it forces a very specific ordering for how elements are combined)  and so we're introducing a new intrinsic that better reflects what LoopVectorize wants to achieve but without the baggage. In this way existing code can pick a format of the intrinsic that can be mapped directly to an `add` instruction if need be.

Reply:
To be specific I'm proposing
```
declare <4 x i32> @llvm.experimental.vector.partial.reduce.add.v4i32.v16i32(<4 x i32>,<16 x i32>)
```
whereby the result and first operand types match, but the second operand differs (perhaps with a restriction that is must have the same or more elements).